### PR TITLE
Delete spectrum annotations

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR fixes delete button for spectrum annotations.

### CODE REVIEW

File: static/js/components/AnnotationsTable.jsx [hunk 1]
```diff
@@ -90,15 +91,27 @@ const AnnotationsTable = ({ annotations, spectrumAnnotations = [] }) => {
...
   };
 
   // Curate data
+  annotations?.forEach((annotation) => {
+    annotation.type = "source";
+  });
+  spectrumAnnotations?.forEach((spectrumAnnotation) => {
+    spectrumAnnotation.type = "spectrum";
+    annotations?.push(spectrumAnnotation);
+  });
```
Comment (guynir42) [inline static/js/components/AnnotationsTable.jsx:113]: I'm not sure this is the best place to mark these. Maybe in the duck that puts them into "annotations" and "spectrumAnnotations"? That way you will have that "type" in any component.

Comment (mcoughlin) [reply static/js/components/AnnotationsTable.jsx:113]: @guynir42 to confirm: the handlers right?

Comment (guynir42) [reply static/js/components/AnnotationsTable.jsx]: In either the handlers or in the duck where you change the store's state, if you want this to only exist in the front end.

Comment (mcoughlin) [reply static/js/components/AnnotationsTable.jsx]: I added it to the handlers.

**Review by guynir42 — APPROVED** [review]
Yeah, that works too!

**Pull request merged**